File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/activities/investigation/evidence/children/solo/views/PhaseTimerControlView.java
package com.tritiumstudios.phasmophobiaevidencetool.android.activities.investigation.evidence.children.solo.views;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageButton;

import com.tritiumstudios.phasmophobiaevidencetool.android.activities.investigation.evidence.children.solo.data.PhaseTimerData;

/**
 * TimerPlayControl class
 *
 * @author TritiumGamingStudios
 */
public class PhaseTimerControlView {

    private final PhaseTimerData phaseTimerData;
    private PhaseTimerView timer;
    private AppCompatImageButton view;

    private int icon_play = 0, icon_pause = 0;

    /**
     * @param timer
     * @param play_pause_view
     * @param icon_play
     * @param icon_pause
     */
    public PhaseTimerControlView(
            PhaseTimerData phaseTimerData,
            PhaseTimerView timer,
            @NonNull AppCompatImageButton play_pause_view,
            int icon_play,
            int icon_pause) {
        this.phaseTimerData = phaseTimerData;

        setTimer(timer);
        setTextView(play_pause_view);

        setPlayBackgroundResource(icon_play);
        setPauseBackgroundResource(icon_pause);

        play_pause_view.setOnClickListener(v -> toggle());

        checkPaused();
    }

    /**
     * @param view
     */
    private void setTextView(AppCompatImageButton view) {
        this.view = view;
    }

    /**
     * @param timer
     */
    public void setTimer(PhaseTimerView timer) {
        this.timer = timer;
    }

    /**
     * @param icon_play
     */
    private void setPlayBackgroundResource(int icon_play) {
        this.icon_play = icon_play;
    }

    /**
     * @param icon_pause
     */
    private void setPauseBackgroundResource(int icon_pause) {
        this.icon_pause = icon_pause;
    }

    /**
     *
     */
    public void checkPaused() {
        if (timer != null && phaseTimerData.isPaused()) {
            view.setImageResource(icon_play);
        } else {
            play();
        }
    }

    /**
     *
     */
    public void pause() {
        view.setImageResource(icon_play);
        timer.pause();
    }

    /**
     *
     */
    public void play() {
        view.setImageResource(icon_pause);
        timer.play();
    }

    /**
     *
     */
    public void toggle() {
        if (phaseTimerData.isPaused())
            play();
        else
            pause();
    }

    public void reset() {
        pause();
    }


}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/evidence/views/PhaseTimerControlView.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.evidence.views

import android.view.View
import androidx.appcompat.widget.AppCompatImageButton
import com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.evidence.data.PhaseTimerModel
import com.TritiumGaming.phasmophobiaevidencepicker.data.viewmodels.EvidenceViewModel

/**
 * TimerPlayControl class
 *
 * @author TritiumGamingStudios
 */
class PhaseTimerControlView {
    private val phaseTimerData: PhaseTimerModel?

    private var timer: PhaseTimerView? = null
    private var view: AppCompatImageButton? = null

    private var icon_play = 0
    private var icon_pause = 0

    constructor(
        evidenceViewModel: EvidenceViewModel,
        timer: PhaseTimerView?,
        play_pause_view: AppCompatImageButton,
        icon_play: Int,
        icon_pause: Int
    ) {
        this.phaseTimerData = evidenceViewModel.phaseTimerData

        setTimer(timer)
        setTextView(play_pause_view)

        setPlayBackgroundResource(icon_play)
        setPauseBackgroundResource(icon_pause)

        play_pause_view.setOnClickListener { v: View? -> toggle() }

        checkPaused()
    }

    constructor(
        phaseTimerData: PhaseTimerModel?,
        timer: PhaseTimerView?,
        play_pause_view: AppCompatImageButton,
        icon_play: Int,
        icon_pause: Int
    ) {
        this.phaseTimerData = phaseTimerData

        setTimer(timer)
        setTextView(play_pause_view)

        setPlayBackgroundResource(icon_play)
        setPauseBackgroundResource(icon_pause)

        play_pause_view.setOnClickListener { v: View? -> toggle() }

        checkPaused()
    }

    private fun setTextView(view: AppCompatImageButton) {
        this.view = view
    }

    fun setTimer(timer: PhaseTimerView?) {
        this.timer = timer
    }

    private fun setPlayBackgroundResource(icon_play: Int) {
        this.icon_play = icon_play
    }

    private fun setPauseBackgroundResource(icon_pause: Int) {
        this.icon_pause = icon_pause
    }

    fun checkPaused() {
        if (timer != null && phaseTimerData!!.isPaused) {
            view!!.setImageResource(icon_play)
        } else {
            play()
        }
    }

    fun pause() {
        view!!.setImageResource(icon_play)
        timer!!.pause()
    }

    fun play() {
        view!!.setImageResource(icon_pause)
        timer!!.play()
    }

    fun toggle() {
        if (phaseTimerData!!.isPaused) play()
        else pause()
    }

    fun reset() {
        pause()
    }
}
